### ISSUE TITLE
hal::temperature_sensor

### ISSUE THREAD

**Issue by kammce** [libhal-google/libhal#137]

**Comment by kammce** [comment]
@MaliaLabor put a comment here

**Comment by MaliaLabor** [comment]
comment

**Comment by kammce** [comment]
Going to remove @MaliaLabor from this assignment as this particular task requires defining how units will be handled in general across the ecosystem and that seems like a pretty daunting first task. I, myself, do not know how units will act going forward so this is something that I should tackle in a separate issue.

**Comment by kammce** [comment]
Was going to simply use nanokelvin for this particular library but it looks like mp-units does not support prefixes for Kelvin. I'm not sure why this is. Going to ask in the issues to see why they choose this and what the utility of no_prefix is in this case. It makes sense disallow milli-foot or mega-inch because a more basic unit would be meters for which there are prefixes for.

**Comment by kammce** [comment]
Looks like this is the reason, its just not supported yet. https://github.com/mpusz/units/pull/232

This interface is basically blocked by this, unless a technical-debt like work around is made.

**Comment by kammce** [comment]
Refer to for details of what was there before #407
